nemo/collections/asr/parts/numba/rnnt_loss/utils/cuda_utils/gpu_rnnt_kernel.py

```diff
@@ -981,21 +983,26 @@ def compute_tdt_alphas_kernel(
         elif u < U:
             # when t == 0, we only consider the non-blank emission.
             if t == 0:
-                alphas[offset + u] = (
-                    alphas[offset + u - 1]  # alpha(t, u - 1)
-                    + logp(
-                        denom, acts, maxT, maxU, alphabet_size, b, t, u - 1, labels[u - 1]
-                    )  # logp of token emission
-                    - sigma  # logit under-normalization
-                    + logp_duration(
-                        duration_acts, maxT, maxU, num_durations, b, t, u - 1, 0
-                    )  # t = 0, so it must be duration = 0. Therefore the last argument passed to logp_duration() is 0.
-                )
+                if durations[0] == 0:
+                    alphas[offset + u] = (
+                        alphas[offset + u - 1]  # alpha(t, u - 1)
+                        + logp(
+                            denom, acts, maxT, maxU, alphabet_size, b, t, u - 1, labels[u - 1]
+                        )  # logp of token emission
+                        - sigma  # logit under-normalization
+                        + logp_duration(
+                            duration_acts, maxT, maxU, num_durations, b, t, u - 1, 0
+                        )  # t = 0, so it must be duration = 0. Therefore the last argument passed to logp_duration() is 0.
+                    )
+                else:
+                    alphas[offset + u] = -INF
 
             # now we have t != 0 and u != 0, and we need to consider both non-blank and blank emissions.
             elif t > 0 and t < T:
```

Comment:
Dumb question: what do we do in cases `t<0` are those just skipped passes on the kernel call?

Reply:
In this case, since `t` was computed earlier at `t = n - u`, it can be negative. And when that happens, the code just skips this part and does nothing.